Common/IRI.hs

```diff
@@ -1091,3 +1092,21 @@ addSuffixToIRI s i = if not $ null $ iriQuery i
                    then i{iriQuery = iriQuery i ++ s}
                   else  
                         i{iriPath  = appendId (iriPath i) (stringToId s)}
+
+-- | Extracts Id from URI
+uriToCaslId :: IRI -> Id
+uriToCaslId urI = 
+ let urS = showIRI urI
+     urS' = concatMap 
+            (\c ->  if isAlpha c ||
+                       isDigit c || 
+                       c == '\'' ||
+                       c == '_'
+                   then [c]
+                   else "_u" ) urS
+ in case urS of
+      x : _ -> 
+         if not (isAlpha x || x == '.') then genName urS' 
```

Comment:
actually, `|| x == '.'` does not make sense here. If the IRI starts with a dot, the dot is replaced with `_u`, and the resulting identifier starts with `_u`, which is not legal in CASL. So the inital dot needs to be kept, but only if a `WORDS` in the CASL sense follows. 
But maybe the whole dot thing is not relevant. So you also could just omit `|| x == '.'`.

Reply:
I think if we omit the test from the code and an IRI should start with a dot, we would get an illegal CASL identifier. So the test should stay. The scheme and the prefix parts of an IRI cannot start with a dot, so the only way to get a dot at the beginning of an IRI is if it's in the IRI path. This should give us a CASL identifier via translation.

Reply:
If we omit the test from the code and an IRI should start with a dot, the dot will be replaced with `_u` and the whole identifier will be prefixed with `gn_`. So `.id` will be translated to `gn__uid`, which is a legal CASL Id.

Alternatively, we could translate `.id` to itself, which also is a legal CASL Id. However, this would require some more code.
 
If we leave things as they are now, `.id` will be translated to `_uid`, which is not a legal CASL Id.

Reply:
try yourself with:
```
ghci Common/IRI.hs
uriToCaslId  nullIRI{iriPath = stringToId ".id"}
```

Reply:
You are right. I removed the test.